src/CLI/usbguard-allow-device.cpp

```diff
@@ -75,14 +76,47 @@ namespace usbguard
     argc -= optind;
     argv += optind;
 
-    if (argc != 1) {
-      showHelp(std::cerr);
-      return EXIT_FAILURE;
+    usbguard::IPCClient ipc(/*connected=*/true);
+
+    if (argc == 1) {
+      id = std::stoul(argv[0]);
+      ipc.applyDevicePolicy(id, Rule::Target::Allow, permanent);
+    }
+    else {
+      /*
+       * Create a rulestring from the arguments
+       */
+      std::string rule_string = argv[0];
+
+      for (int i = 1; i < argc; i++) {
+        rule_string.append(" ").append(argv[i]);
+      }
+
+      usbguard::Rule rule;
+      try {
+        rule = Rule::fromString(rule_string);
+      }
+      catch (...) {
+        showHelp(std::cerr);
+        return EXIT_FAILURE;
+      }
+
+      /*
+       * Create a ruleset which will contain the requested rule
+      */
+      auto my_rule_set = std::dynamic_pointer_cast<RuleSet>(std::make_shared<MemoryRuleSet>(nullptr));
+      my_rule_set->appendRule(rule);
+
+      for (auto rule_device : ipc.listDevices("match")) {
```

Comment:
I am not sure if the **allow|block|reject-device** commands should affect all the devices or just a few of them, e.g. those which are "allowed", "blocked".
My question is whether I should replace
`for (auto rule_device : ipc.listDevices("match")) {`
with
`for (auto rule_device : ipc.listDevices(argv[0])) {`
@dkopecek

Reply:
I'm not sure what listDevices(argv[0]) means but I think you can use the match rule constructed from the command line argument directly as a query in listDevices(). But listing all devices and applying the rule only on the matching ones is equivalent. The difference is only which side (cli or daemon) is doing the matching.

Reply:
Sorry, my question was not well formed. 
I choose the first option - the CLI is doing the matching.

Reply:
Ok